url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/imex/FisExporter.java

```diff
@@ -0,0 +1,527 @@
+            return String.format("'%s':'rectmf',[%s]", term.getName(),
+                    Op.join(" ", t.getStart(), t.getEnd()));
+        }
+        if (term instanceof SigmoidDifference) {
+            SigmoidDifference t = (SigmoidDifference) term;
+            return String.format("'%s':'dsigmf',[%s]", term.getName(),
+                    Op.join(" ", t.getRising(), t.getLeft(),
+                            t.getFalling(), t.getRight()));
+        }
+        if (term instanceof Sigmoid) {
+            Sigmoid t = (Sigmoid) term;
+            return String.format("'%s':'sigmf',[%s]", term.getName(),
+                    Op.join(" ", t.getSlope(), t.getInflection()));
+        }
+        if (term instanceof SigmoidProduct) {
+            SigmoidProduct t = (SigmoidProduct) term;
+            return String.format("'%s':'psigmf',[%s]", term.getName(),
+                    Op.join(" ", t.getRising(), t.getLeft(),
+                            t.getFalling(), t.getRight()));
+        }
+        if (term instanceof SShape) {
+            SShape t = (SShape) term;
+            return String.format("'%s':'smf',[%s]", term.getName(),
+                    Op.join(" ", t.getStart(), t.getEnd()));
+        }
+        if (term instanceof Trapezoid) {
+            Trapezoid t = (Trapezoid) term;
+            return String.format("'%s':'trapmf',[%s]", term.getName(),
+                    Op.join(" ", t.getA(), t.getB(), t.getC(), t.getD()));
+        }
+        if (term instanceof Triangle) {
+            Triangle t = (Triangle) term;
+            return String.format("'%s':'trimf',[%s]", term.getName(),
+                    Op.join(" ", t.getA(), t.getB(), t.getC()));
+        }
+        if (term instanceof ZShape) {
+            ZShape t = (ZShape) term;
+            return String.format("'%s':'zmf',[%s]", term.getName(),
+                    Op.join(" ", t.getStart(), t.getEnd()));
+        }
+        throw new RuntimeException(String.format("[export error] "
+                + "term of class <%s> not supported", term.getClass().getName()));
+    }
+
+    public String toString(Defuzzifier defuzzifier) {
+        if (defuzzifier == null) {
+            return "";
+        }
+        if (defuzzifier instanceof Centroid) {
+            return "centroid";
+        }
+        if (defuzzifier instanceof Bisector) {
+            return "bisector";
+        }
+        if (defuzzifier instanceof SmallestOfMaximum) {
+            return "som";
+        }
+        if (defuzzifier instanceof LargestOfMaximum) {
+            return "lom";
+        }
+        if (defuzzifier instanceof MeanOfMaximum) {
+            return "mom";
+        }
+        if (defuzzifier instanceof WeightedAverage) {
+            return "wtaver";
+        }
+        if (defuzzifier instanceof WeightedSum) {
+            return "wtsum";
+        }
+        return defuzzifier.getClass().getSimpleName();
+    }
+
+    public String toString(Norm norm) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.